ui/raidboss/emulator/data/PopupTextAnalysis.ts

```diff
@@ -0,0 +1,289 @@
+import EmulatorCommon, { DataType } from '../EmulatorCommon';
+import StubbedPopupText from '../overrides/StubbedPopupText';
+import LineEvent from './network_log_converter/LineEvent';
+import { LooseTrigger, MatchesAny } from '../../../../types/trigger';
+import { UnreachableCode } from '../../../../resources/not_reached';
+import { TriggerHelper, Text, TextText, ProcessedTrigger } from '../../popup-text';
+import { EventResponses, LogEvent } from '../../../../types/event';
+
+type ResolverFunc = () => void;
+
+export type EmulatorLogEvent = LogEvent & {
+  detail: {
+    logs: LineEvent[];
+  };
+}
+
+export type EmulatorNetworkLogEvent = EventResponses['LogLine'] & {
+  detail: {
+    logs: LineEvent[];
+  };
+}
+
+interface ResolverStatus {
+  responseType?: string;
+  responseLabel?: string;
+  initialData: DataType;
+  finalData?: DataType;
+  condition?: boolean;
+  response?: undefined;
+  result?: string;
+  delay?: number;
+  suppressed: boolean;
+  executed: boolean;
+  promise?: Promise<void>;
+}
+
+type EmulatorTriggerHelper = TriggerHelper & {
+  resolver?: Resolver;
+};
+
+class Resolver {
+  private promise?: Promise<void>;
+  private run?: ResolverFunc;
+  private delayUntil?: number;
+  private final?: ResolverFunc;
+  private delayPromise?: Promise<void>;
+  private delayResolver?: (value: void | PromiseLike<void>) => void;
```

Comment:
```suggestion
  private delayResolver?: () => void;
```

`delayResolver` is only ever called with no parameters, so I would have expected something more like this suggestion above.  Am I missing something?

Reply:
This was a leftover from the first pass at conversion and should have been `ResolverFunc` instead. Fixed, thanks.